app/validators/checksum_validator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+class ChecksumValidator < ActiveModel::Validator
+  include Rails.application.routes.url_helpers
+  def validate(record)
+    @record = record
+    @uploaded_file = @record.file
+    @duvs_obj = DuplicateUploadVerificationService.new(@uploaded_file)
+    if !@duvs_obj.duplicates.empty?
+      record.errors.add(:checksum, in_solr_error_message)
+    elsif Sufia::UploadedFile.begun_ingestion.map(&:checksum).include?(@record.checksum)
```

Comment:
this looks like it can be refactored to be less taxing. I would just chain a where onto the begun_ingestion scope and see if is empty or not.